### DISCUSSION TITLE
Guidance on using pg_auto_failover  with multi node Citus DB

### DISCUSSION THREAD

**Discussion by dodyg** [hapostgres/pg_auto_failover#636] [Q&A] [answered]
Sorry if this sounds like a stupid question but I am researching HA options at the moment for PostgreSQL although I am just a casual user.

Is there any guidance written somewhere on how to use pg_auto_failover with multi-node Citus? Citus  has a section about [node failure](http://docs.citusdata.com/en/v10.0/admin_guide/cluster_management.html#worker-node-failures) that recommend Streaming replication to a standby.  It's not clear for me whether this option will enable that the Citus to just work without a manual intervention when a node is down.

**Comment by s4ke** [comment]
I am in the same situation right now, but as far as I can tell the following should work, but I have not tried it yet:

1. Create HA cluster for coordinator node
2. Create HA cluster for data node 1
3. Create HA cluster for data node 2
4. ...
5. Create cluster https://github.com/citusdata/citus#install-citus-on-multiple-nodes

Essentially think of any of the clusters as a single node in the citus design. pg_auto_failover supports citus formations, so that might be helpful as well.

Am I right, @DimCitus ?

**Reply by s4ke** [reply]
see https://github.com/citusdata/pg_auto_failover/issues/362

**Comment by s4ke** [comment, marked as answer]
With the recent release of 2.0 and citus support, this is worth checking out again for you maybe @dodyg

**Reply by DimCitus** [reply]
Please see the new [Citus HA Tutorial](https://pg-auto-failover.readthedocs.io/en/main/citus-quickstart.html) indeed.
